File: src/LeaveManagement.Api/Domain/LeaveRequests/LeaveRequest.cs
using StarterKit.Shared.Entities;

namespace StarterKit.LeaveManagement.Api.Domain.LeaveRequests;

public class LeaveRequest : AuditableEntity
{
    public string UserId { get; set; } = null!;

    public string EmployeeId { get; set; } = null!;

    public LeaveType LeaveType { get; set; }

    public DateTimeOffset StartDate { get; set; }

    public DateTimeOffset EndDate { get; set; }

    public string? Reason { get; set; }

    public LeaveRequestStatus Status { get; set; } = LeaveRequestStatus.Pending;

    public string? ApprovalRequestId { get; set; }
}

File: clients/admin/src/modules/leave-requests/types/leave-request.ts
/** Mirrors `LeaveManagement.Contracts/LeaveRequests/LeaveType.cs` — serialized by name. */
export enum LeaveType {
  Annual = "Annual",
  Sick = "Sick",
  Unpaid = "Unpaid",
  Other = "Other",
}

/** Mirrors `LeaveManagement.Contracts/LeaveRequests/LeaveRequestStatus.cs` — serialized by name. */
export enum LeaveRequestStatus {
  Pending = "Pending",
  Approved = "Approved",
  Rejected = "Rejected",
  Cancelled = "Cancelled",
}

export interface LeaveRequestDto {
  id: string;
  userId: string;
  employeeId: string;
  leaveType: LeaveType;
  startDate: string;
  endDate: string;
  reason?: string | null;
  status: LeaveRequestStatus;
  approvalRequestId?: string | null;
  created: string;
}

export interface ApproverCandidateDto {
  employeeId: string;
  userId: string;
  name: string;
}

export interface CreateLeaveRequestPayload {
  leaveType: LeaveType;
  startDate: string;
  endDate: string;
  reason?: string;
  approverEmployeeId: string;
}

/** `approverEmployeeId` is only meaningful (and required) on the self-service resubmission path —
 * a manage-scoped edit is metadata-only and never touches Approval. */
export interface UpdateLeaveRequestPayload {
  leaveType: LeaveType;
  startDate: string;
  endDate: string;
  reason?: string;
  approverEmployeeId?: string;
}

export interface LeaveRequestSearchParams {
  leaveType?: LeaveType;
  status?: LeaveRequestStatus;
  /** Only honored server-side for callers with `leave.requests.manage`. */
  employeeId?: string;
  pageNumber?: number;
  pageSize?: number;
}
